helix-view/src/editor.rs

```diff
@@ -449,7 +449,8 @@ pub struct Editor {
 
     pub syn_loader: Arc<syntax::Loader>,
     pub theme_loader: Arc<theme::Loader>,
-
+    // last_theme is currently just used for theme previews
+    pub last_theme: Option<String>,
```

Comment:
Would it be better if we just store it as `Option<Box<Theme>>` rather than `String`, that way we don't have to keep `name`.

And instead of setting `last_theme` over the lifetime of the application, we can just set it when user starts previewing a theme, that way only during preview we allocate. Once preview is done, we replace current `theme` with `last_theme`, that way we don't have to store extra stuff when not previewing and easier to manage.